LiteCore/Query/QueryParser.cc

```diff
@@ -1164,30 +1166,37 @@ namespace litecore {
 
     namespace {
         const char* const metaKeys[][2] = {
-            {"id",         "meta-id@@@"},
-            {"sequence",   "meta-sequence@@@"},
-            {"deleted",    "meta-deleted@@@"},
-            {"expiration", "meta-expiration@@@"},
-            {"revisionID", "meta-revisionID@@@"}
+            {"id",         "f1rK.b3xI2bA6" "id"},
+            {"sequence",   "f1rK.b3xI2bA6" "sequence"},
+            {"deleted",    "f1rK.b3xI2bA6" "deleted"},
+            {"expiration", "f1rK.b3xI2bA6" "expiration"},
+            {"revisionID", "f1rK.b3xI2bA6" "revisionID"}
```

Comment:
Was there something wrong with the old way or does this add performance or something?

Reply:
not wrong with the old way. Basically I was looking for something that I can be sure that, ""f1rK.b3xI2bA6" was put there by me. Previously, I test "meta-key" at the beginning and "@@@" at the end. Now, I only test the beginning. Assumption here is no one will choose db alias starting by "f1rK.b3xI2bA6". 
The real improvement is that I avoided parsing JSON into a Value structure.

Reply:
While that assumption is probably fine any assumption that a user will not do something makes me nervous.  Is there a way to catch the case that this happens?  Perhaps even hard-coding in this value to be rejected as an alias?

Reply:
This is the story.
"Meta(alias)" will be translated to '["meta", "alias"]'. Users don't spell out all the meta keys. 
Then I played a trick, and translating it to a dictionary of '{"id": ["meta", "f1rK.b3xI2bA6idalias"], "deleted": ["meta", "f1rK.b3xI2bA6deletedalias"], ...}'. Then, by the current parser machine, the "meta" operator will be called again. At that time, I write to sql stream the function to get the meta key from sql table.
Apparently, I made an assumption here that users will never choose database alias like "f1rK.b3xI2bA6idalias". Basically, I overloaded operator, ["meta", ...]
To avoid the assumption, I can make a new internal operator, "meta-internal". This will make things crystal clear. Would you recommend this route?

Reply:
This is _really_ convoluted and it's taking me a long time to figure out. As I understand it, you're handling `["META"]` by generating a Dict on the fly that contains META expressions to get each magic property, then feeding that back into the QueryParser to generate SQL.

Why not just emit a call to `dict_of`, using `writeMetaPropertyGetter` to write each of the values?

Reply:
Yes, you're right: I meant to feed it back to QueryParser. This way, I can separate the dictionary part and the part to get the meta properties from the sqlite's table. And, I don't have to worry about how we turning a dictionary in query into dictionary in the result, and how to extract a field/property from the dictionary result.